dev/breeze/src/airflow_breeze/commands/release_candidate_command.py

```diff
@@ -523,14 +534,17 @@ def remove_old_releases(version, repo_root):
 
 @release_management.command(
     name="prepare-airflow-tarball",
-    help="Prepare airflow's source tarball.",
+    help="Prepare airflow's or airflow distribution source tarball.",
 )
 @click.option(
     "--version", required=True, help="The release candidate version e.g. 2.4.3rc1", envvar="VERSION"
 )
+@click.option(
+    "--distribution-name", default="airflow", help="The distribution name, airflow, task-sdk, airflowctl"
```

Comment:
NIT 1: We could turn it it into "BetterChoice" type of field and pass the list o choices (will validate and autocomplete)

NIT 2: We should add "providers" as well.

Reply:
Would be great also to update release processes for Airflow, Providers, Task-SDK, airflowctl to add the step to produce it :)

Reply:
Thanks for your comments, Jarek! :) I addressed them.

Hope I made the changes to the release processes correctly, please double-check :) I have added a tag to specify to `git archive`, and if it is passed, the distribution commands also generate a source tarball. Or still `breeze release-management prepare-airflow-tarball` can be used. 

The `tag` logic was also added in general because sometimes a version without rc does not correspond to a tag, and since only `airflow-core` version matching with tag names such as `3.0.6`, this makes other distributions a bit not testable and working in a couple of cases. This stabilises and lets say we can give `airflowctl/1.0.0b1,` but still can create 1.0.0 source tar.